pip/src/interop.rs

```diff
@@ -79,18 +80,46 @@ where
     }
 }
 
-/// This call while exported is not intended to be used directly by the user.
-/// It is intended to be used by the Python wrapper which will handle the
-/// callbacks and other Python specific details.
+/// Runs the given OpenQASM program for the given number of shots.
+/// Each shot uses an independent instance of the simulator.
+///
+/// Note:
+///     This call while exported is not intended to be used directly by the user.
+///     It is intended to be used by the Python wrapper which will handle the
+///     callbacks and other Python specific details.
+///
+/// Args:
+///     source (str): The OpenQASM source code to execute.
+///     output_fn (Callable[[Output], None]): The function to handle the output of the execution.
+///     noise: The noise to use in simulation.
+///     read_file (Callable[[str], Tuple[str, str]]): The function to read a file and return its contents.
+///     list_directory (Callable[[str], List[Dict[str, str]]]): The function to list the contents of a directory.
+///     resolve_path (Callable[[str, str], str]): The function to resolve a path given a base path and a relative path.
+///     fetch_github (Callable[[str, str, str, str], str]): The function to fetch a file from GitHub.
+///     **kwargs: Additional keyword arguments to pass to the execution.
+///       - target_profile (TargetProfile): The target profile to use for execution.
+///       - name (str): The name of the circuit. This is used as the entry point for the program. Defaults to 'program'.
+///       - search_path (str): The optional search path for resolving imports.
+///       - output_semantics (OutputSemantics, optional): The output semantics for the compilation.
+///       - shots (int): The number of shots to run the program for. Defaults to 1.
+///       - seed (int): The seed to use for the random number generator.
+///
+/// Returns:
+///     Any: The result of the execution.
+///
+/// Raises:
+///     QasmError: If there is an error generating, parsing, or analyzing the OpenQASM source.
+///     QSharpError: If there is an error interpreting the input.
 #[pyfunction]
 #[allow(clippy::too_many_arguments)]
 #[pyo3(
-    signature = (source, callback=None, read_file=None, list_directory=None, resolve_path=None, fetch_github=None, **kwargs)
+    signature = (source, callback=None, noise=None, read_file=None, list_directory=None, resolve_path=None, fetch_github=None, **kwargs)
 )]
-pub fn run_qasm(
+pub fn run_qasm_program(
```

Comment:
nitpick: does this need to be `pub`? The others are `pub(crate)` which seems to work fine with pyO3 and how it's used within this crate.

```suggestion
pub(crate) fn run_qasm_program(
```